corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -518,23 +518,129 @@ hqDefine('app_manager/js/app_manager', [
+                                $caseType.append(option);
+                            }
+                        });
+                    },
+                });
 
-            $('#add-new-module-modal').modal('hide');
+                $caseType.on('change', function () {
+                    var value = $(this).val();
+                    var valueNoSpaces = value ? value.replace(/ /g, '_') : '';
+                    var $formGroup = $(this).closest('.form-group');
+                    var help = $('#new_case_type_help');
+                    var $createBtn = $('#case-type-create-btn');
+
+                    // Reset error states
+                    $formGroup.removeClass('has-error');
+                    help.text(gettext("* You can change the Case Type later by going to the menu settings page"));
+                    $createBtn.prop('disabled', false);
+
+                    if (!valueNoSpaces) {
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (!valueNoSpaces.match(/^[\w-]*$/g)) {
+                        $formGroup.addClass('has-error');
+                        help.text(gettext("Case types can only include the characters a-z, 0-9, '-' and '_'"));
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (valueNoSpaces === 'commcare-user') {
+                        $formGroup.addClass('has-error');
+                        help.text(gettext("'commcare-user' is a reserved case type. Please change the case type"));
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (self.deprecatedCaseTypes().includes(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        help.text(gettext("This case type has been deprecated in the Data Dictionary."));
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+                });
+            }
+        });
+
+        // Handle "Create Application" button click
+        $('#case-type-create-btn').on('click', function () {
+            google.track.event("Added Case List Menu");
+            kissmetrix.track.event("Added Case List Menu");
+
+            var $caseTypeInput = $('#new_case_type');
+            var value = $caseTypeInput.val();
+            var valueNoSpaces = value.replace(/ /g, '_');
+
+            // If validation passes, update the value and submit the form
+            if (value !== valueNoSpaces) {
+                $caseTypeInput.val(valueNoSpaces);
+            }
+
+            var $form = $('#new-module-form');
+            $form.append('<input type="hidden" name="case_type" value="' + valueNoSpaces + '" />');
+
+            $('.new-module-icon').removeClass().addClass("fa fa-refresh fa-spin");
```

Comment:
Does the `removeClass` call without params remove all classes?

Reply:
Yes, `<i class="new-module-icon fa fa-plus"></i>` will become `<i class="fa fa-refresh fa-spin">`. I think it is fine to remove `new-module-icon` for a few seconds since we will reload the page when new module is loaded..